ddtrace/tracer.py

```diff
@@ -283,6 +287,8 @@ def start_span(self, name, child_of=None, service=None, resource=None, span_type
             # service(s) that may have been added.
             self._update_dogstatsd_constant_tags()
 
+        self.thread_to_span[compat.thread_get_ident()] = span
```

Comment:
what if we are in gevent/asyncio?

Reply:
should we be keeping track of all "opened" contexts instead?

Reply:
> what if we are in gevent/asyncio?

Then there's a single thread (unless you create your own native threads yourself by bypassing it) and that's it. This is not trying to track contextes.

> should we be keeping track of all "opened" contexts instead?

 I don't think it matters: it's a weakref dict, so there's just a need to check the `finished` attribute of the span to know if it's done or not. And if it's done and flushed and not ref'ed, it'll disappear from the dict.

Reply:
👍